jobs: mark job as running after adopting

spaskob opened the pull request:
We could adopt a job and run it on a node without changing its status to
running.

Release note (bug fix): job can be running but shown in state pending.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43814)
<!-- Reviewable:end -->

dt commented on githooks/prepare-commit-msg:
```diff
@@ -1,7 +1,7 @@
 #
 # Prepare the commit message by adding a release note.
-
+set -x
```
was this supposed to go in this PR?

spaskob committed:
jobs: mark job as running after adopting

We may adopt a job and run it on a node without changing its status to
running.

Release note (bug fix): job can be running but shown in state pending.

spaskob commented:
bors r+

knz commented:
Folk this release note is broken. 

@jseldess the release note text for this PR in 20.1-beta1 is unreadable/unusable. Can you coordinate having spas provide an alternate/better text?

knz commented:
For reference: https://wiki.crdb.io/wiki/spaces/CRDB/pages/186548364/Release+notes
